language/testing-infra/transactional-test-runner/src/vm_test_harness.rs

```diff
@@ -36,13 +37,19 @@ use move_vm_runtime::{
 };
 use move_vm_test_utils::{gas_schedule::GasStatus, InMemoryStorage};
 use once_cell::sync::Lazy;
+use std::marker::PhantomData;
 
 const STD_ADDR: AccountAddress = AccountAddress::ONE;
 
-struct SimpleVMTestAdapter<'a> {
+trait RecordWriteSet {
+    fn record_writeset() -> bool;
+}
+
```

Comment:
This feels a bit excessive. But I get why you had to do it. Maybe a const generic would work a bit better... but some more thoughts on this below.